src/app/api/v1/user/remove-email/route.ts

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { getToken } from "next-auth/jwt";
+import { NextRequest, NextResponse } from "next/server";
+import AppConstants from "../../../../../appConstants";
+
+import {
+  getSubscriberByEmail,
+  deleteResolutionsWithEmail,
+} from "../../../../../db/tables/subscribers";
+import {
+  removeOneSecondaryEmail,
+  getEmailById,
+} from "../../../../../db/tables/emailAddresses.js";
+import { getL10n } from "../../../../functions/server/l10n";
+
+interface EmailDeleteRequest {
+  emailId: string;
+}
+
+export async function POST(req: NextRequest) {
+  const l10n = getL10n();
+  const token = await getToken({ req });
+
+  if (token) {
+    try {
+      const { emailId }: EmailDeleteRequest = await req.json();
+      const subscriber = await getSubscriberByEmail(token.email);
+      const existingEmail = await getEmailById(emailId);
+
+      if (existingEmail?.subscriber_id !== subscriber.id) {
+        return NextResponse.json({
+          success: false,
+          status: 400,
+          message: l10n.getString("error-not-subscribed"),
+        });
+      }
+
+      await removeOneSecondaryEmail(emailId);
+      deleteResolutionsWithEmail(
+        existingEmail.subscriber_id,
+        existingEmail.email
+      );
+      return NextResponse.redirect(
+        AppConstants.SERVER_URL + "/user/settings",
+        301
+      );
+    } catch (e) {
+      console.error(e);
+      return NextResponse.json({ success: false }, { status: 500 });
```

Comment:
Q: Does this need a `message`?

Reply:
I don't think so, since we don't know what the cause of this error is - I don't think we want to expose internal eror messages to the outside world.